lib/uploadcare/api/file_storage_api.rb

```diff
@@ -0,0 +1,29 @@
+module Uploadcare
+  module FileStorageApi
+    BATCH_SIZE = 100
+
+    def store_files(objects)
+      in_batches(objects) { |batch| put "/files/storage/", to_uuids(batch) }
+    end
+
+    def delete_files(objects)
+      in_batches(objects) { |batch| delete "/files/storage/", to_uuids(batch) }
+    end
+
+    private
+
+    def in_batches(enum)
+      enum.each_slice(BATCH_SIZE) { |batch| yield batch }
+    end
+
+    def to_uuids(objects)
+      objects.map do |object|
+        case object
+        when Uploadcare::Api::File then object.uuid
+        when /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/ then object
+        else raise(ArgumentError, "Unable to convert object to uuid: #{object}")
```

Comment:
I see two ways to do so:

1. We limit max input size to 100. One `#store_files` / `#delete_files` call - one request. Return value = api response.  `#store_files` / `#delete_files` methods remain simple, which is good.

    In this case, storage/deletion of FileList becames problematic because we can't tell how much files are there in total before they all will be loaded. To handle this we can add separate methods, `#store` and `#delete`, to FileList class. They'll use `#store_files` and `#delete_files` internaly, which will be called multiple times if there is more than 100 files in a FileList. To provide user with API responses, we can make this methods to acceps a block in which we'll pass each response:

```ruby
list = api.file_list

list.delete do |response| # this block will be called for each internat #delete_files request
  if response['problems'].any?
    # ...
  end
end
```

2. We add the same block as above to `#store_files` and `#delete_files` methods, passing each API response into it. In this case those methods became smart.